<a href="https://colab.research.google.com/github/mylinh030203/AI/blob/main/Analysis_and_Prediction_Diabetes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import warnings
warnings.filterwarnings("ignore")

import numpy as np
import pandas as pd
import random

import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
plt.style.use('ggplot')

#Kích thước của một hình ảnh đơn
onefig_size = (10,4)
#Kích thước của nhiều hình ảnh
multifigs_size = (12,4)

Dataset overview
Dataset bao gồm 253681 hàng, 22 cột
( kết quả từ khảo sát https://www.cdc.gov/pcd/issues/2019/pdf/19_0109.pdf)
* `Diabetes_012` : các nhãn; 0 = không bị tiểu đường,  1 =  tiền tiểu đường,  2 = tiểu đường
* `HighBP` : blood pressure (huyết áp); 0 = no high BP, 1 = high BP
* `HighChol` : cholesterol; 0 = no high cholesterol, 1 = high cholesterol
* `CholCheck` : Kiểm tra cholesterol trong vòng 5 năm; 0 = no cholesterol check in 5 years, 1 = yes cholesterol check in 5 years
* `BMI` : Body Mass Index (chỉ số khối cơ thể)
* `Smoker	` :Hút ít nhất 100 điếu thuốc lá(5 bao thuốc) trong đời;  0 = no, 1 = yes
* `Stroke	` : Từng bị đọt quỵ; 0 = no, 1 = yes
* `HeartDiseaseorAttack	` : bệnh tim mạch vành (CHD) hoặc nhồi máu cơ tim (MI); 0 = no, 1 = yes
* `PhysActivity	` : hoạt động thể chất trong 30 ngày qua - không bao gồm công việc; ; 0 = no, 1 = yes
* `Fruits	` : Ăn trái cây 1 lần trở lên mỗi ngày; 0 = no, 1 = yes
* `Veggies	` : Ăn rau 1 lần trở lên mỗi ngày; 0 = no, 1 = yes
* `HvyAlcoholConsump	` : Người nghiện rượu nặng (đàn ông trưởng thành uống hơn 14 ly mỗi tuần và phụ nữ trưởng thành uống hơn 7 ly mỗi tuần);  0 = no, 1 = yes
* `AnyHealthcare	` : Có bất kỳ loại bảo hiểm chăm sóc sức khỏe nào, bao gồm bảo hiểm y tế, các chương trình trả trước như HMO, v.v; 0 = no, 1 = yes
* `NoDocbcCost	` : Có thời điểm nào trong 12 tháng qua bạn cần đến gặp bác sĩ nhưng không thể đi khám vì lý do chi phí?; 0 = no, 1 = yes
* `GenHlth	` : nhìn chung sức khỏe rất tốt: 1 = excellent, 2 = very good, 3 = good, 4 = fair, 5 = poor
* `MentHlth	` : Sức khỏe tâm thần của bạn, bao gồm căng thẳng, trầm cảm và các vấn đề về cảm xúc, sức khỏe tâm thần của bạn ***không tốt*** trong bao nhiêu ngày trong 30 ngày qua
* `PhysHlth	` : sức khỏe thể chất của bạn, bao gồm bệnh tật và chấn thương thể chất, trong 30 ngày qua sức khỏe thể chất của bạn ***không tốt*** trong bao nhiêu ngày
* `DiffWalk	` :  gặp khó khăn nghiêm trọng khi đi bộ hoặc leo cầu thang;  0 = no, 1 = yes
* `Sex	` : 0 = female, 1 = male
* `Age	` : tuồi của người tham gia khảo sát; 1 = 18-24,( 2 = 25-29, 3= 30-34, 4=35-39, 5 = 40-44, 6 = 45-49, 7 = 50-54, 8 =55-59), 9 = 60-64,( 10= 65-69,11=70-74,12=75-89),  13 = 80 trở lên
* `Education	` : 1 = Chưa bao giờ đi học hoặc chỉ đi học mẫu giáo, 2 = Lớp 1 đến lớp 8 (Tiểu học), 3 = Lớp 9 đến lớp 11 (Một số trường trung học), 4 = Lớp 12 hoặc GED (Tốt nghiệp trung học), 5 = Đại học 1 năm đến 3 năm (Một số trường cao đẳng hoặc kỹ thuật), 6 = Đại học 4 năm trở lên (Tốt nghiệp đại học)
* `Income	` : Thu nhập của người tham gia khảo sát; 1 = dưới $10.000, 5 = dưới  $35.000, 8 = $75.000 trở lên


In [6]:
from google.colab import drive
drive.mount('/content/drive')
# Đọc dữ liệu từ file CSV
# df = pd.read_csv('diabetes.csv')
import os
path="//content//drive//MyDrive//HocMay//"
os.chdir(path)
%pwd

data = pd.read_csv('diabetes2015.csv', delimiter=',')
#5 dong dau, tail() 5 dong cuoi
data.head()

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


Diabetes_012  HighBP  HighChol  CholCheck   BMI  Smoker  Stroke  \
0           0.0     1.0       1.0        1.0  40.0     1.0     0.0   
1           0.0     0.0       0.0        0.0  25.0     1.0     0.0   
2           0.0     1.0       1.0        1.0  28.0     0.0     0.0   
3           0.0     1.0       0.0        1.0  27.0     0.0     0.0   
4           0.0     1.0       1.0        1.0  24.0     0.0     0.0   

   HeartDiseaseorAttack  PhysActivity  Fruits  ...  AnyHealthcare  \
0                   0.0           0.0     0.0  ...            1.0   
1                   0.0           1.0     0.0  ...            0.0   
2                   0.0           0.0     1.0  ...            1.0   
3                   0.0           1.0     1.0  ...            1.0   
4                   0.0           1.0     1.0  ...            1.0   

   NoDocbcCost  GenHlth  MentHlth  PhysHlth  DiffWalk  Sex   Age  Education  \
0          0.0      5.0      18.0      15.0       1.0  0.0   9.0        4.0   
1          1.0      3.0       0.0       0.0       0.0  0.0   7.0        6.0   
2          1.0      5.0      30.0      30.0       1.0  0.0   9.0        4.0   
3          0.0      2.0       0.0       0.0       0.0  0.0  11.0        3.0   
4          0.0      2.0       3.0       0.0       0.0  0.0  11.0        5.0   

   Income  
0     3.0  
1     1.0  
2     8.0  
3     6.0  
4     4.0  

[5 rows x 22 columns]

Thông tin của data

In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 253680 entries, 0 to 253679
Data columns (total 22 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   Diabetes_012          253680 non-null  float64
 1   HighBP                253680 non-null  float64
 2   HighChol              253680 non-null  float64
 3   CholCheck             253680 non-null  float64
 4   BMI                   253680 non-null  float64
 5   Smoker                253680 non-null  float64
 6   Stroke                253680 non-null  float64
 7   HeartDiseaseorAttack  253680 non-null  float64
 8   PhysActivity          253680 non-null  float64
 9   Fruits                253680 non-null  float64
 10  Veggies               253680 non-null  float64
 11  HvyAlcoholConsump     253680 non-null  float64
 12  AnyHealthcare         253680 non-null  float64
 13  NoDocbcCost           253680 non-null  float64
 14  GenHlth               253680 non-null  float64
 15  

## Data Cleaning
### Missing Data
Kiểm tra xem dữ liệu bị nhỡ bao nhiêu


In [8]:
data.isnull().sum()*100/data.shape[0]

Diabetes_012            0.0
HighBP                  0.0
HighChol                0.0
CholCheck               0.0
BMI                     0.0
Smoker                  0.0
Stroke                  0.0
HeartDiseaseorAttack    0.0
PhysActivity            0.0
Fruits                  0.0
Veggies                 0.0
HvyAlcoholConsump       0.0
AnyHealthcare           0.0
NoDocbcCost             0.0
GenHlth                 0.0
MentHlth                0.0
PhysHlth                0.0
DiffWalk                0.0
Sex                     0.0
Age                     0.0
Education               0.0
Income                  0.0
dtype: float64

Thấy rằng không có dữ liệu bị thiếu nên chuyển qua bước phân tích để có thêm nhiều kiến thức

## Phân tích dữ liệu
Dữ liệu đã được làm sạch tổng cộng có 253680 dòng, bao gồm:
* Biến phân loại: `HighBP `, `HighChol`, `CholCheck`, `BMI `, `Smoker `, `Stroke `, `HeartDiseaseorAttack `, `PhysActivity`, `Fruits`, `Veggies`, `HvyAlcoholConsump`, ` AnyHealthcare`, `NoDocbcCost`, `GenHlth`, `DiffWalk`,`Sex `, `Age`,`Education`, `Income`
* Biến số: , `BMI `, `MentHlth`, `PhysHlth`